### DISCUSSION TITLE
Upgrading from 2.6.6?

### DISCUSSION THREAD

**Discussion by jeffkruczynski** [jasonacox/Powerwall-Dashboard#201] [Q&A] [answered]
Hello,
I am currently running 2.6.6 and would like to upgrade.
It looks like improvements have been made to upgrade.sh since then, so I'm wondering: should I use the upgrade.sh installed with 2.6.6, or is there value in using a newer version of that script? (if that would even work)
Thanks!

**Comment by BJReplay** [comment, marked as answer]
The upgrade script gets the latest upgrade script to run the upgrade.  You should see this in your version - [https://github.com/jasonacox/Powerwall-Dashboard/blob/main/upgrade.sh#L34](https://github.com/jasonacox/Powerwall-Dashboard/blob/main/upgrade.sh#L34) grabs the latest version to run the upgrade:

```
# Because this file can be upgrade, don't use it to run the upgrade
if [ "$0" != "tmp.sh" ]; then
    # Grab latest upgrade script from github and run it
    curl -sL --output tmp.sh https://raw.githubusercontent.com/jasonacox/Powerwall-Dashboard/main/upgrade.sh
    exec bash tmp.sh upgrade
fi
```

This means you should be fine to run your version.

**Reply by jeffkruczynski** [reply]
Ah — I hadn't seen that. Thank you!
